### PR TITLE
New --patchline argument

### PR DESCRIPTION
**Pull request by LeagueOfPoro**
Added support for `--patchline pbe` argument as a working alternative to `--cdn`.

### CODE REVIEW

File: cdragontoolbox/__main__.py [hunk 1]
```diff
@@ -74,6 +74,8 @@ def parse_storage_args(parser, args) -> Storage:
             conf['cdn'] = cdn
         elif cdn != 'default':
             parser.error("--cdn is only supported for 'rads' storage")
+        if conf['type'] == 'patcher':
+            conf['patchline'] = args.patchline
```
Comment (benoitryder) [inline cdragontoolbox/__main__.py:78]: An error should be thrown if `--patchline` is set and storage type is not `patcher` (as for `--cdn`).
This is easy to do if `--patchline` defaults to `None` as suggested.

**Review by benoitryder — CHANGES_REQUESTED** [review]
Thanks for your PR! Several people had issues with that in the past so this will definiteily be helpful. :)

**Comment by benoitryder** [comment]
Keeping `--patchline` is better IMHO, because it's more generic and matches directly the storage JSON format.

And you can use the following to have same default/fallback behavior:
```py
    storage_parser.add_argument('--patchline', choices=['live', 'pbe'], default=None,
                                help="use PBE patchline")
```
```py
        if args.patchline is not None:
            if conf['type'] == 'patcher':
                conf['patchline'] = args.patchline
            else:
                parser.error("--patchline is only supported for 'patcher' storage")
```

**Comment by LeagueOfPoro** [comment]
I would argue --pbe makes for a more concise command. However, matching the JSON config makes sense. 
Changed

**Pull request merged**